spyderlib/plugins/configdialog.py

```diff
@@ -903,77 +916,433 @@ def _save_lang(self):
+        self.load_from_conf()
+
+        if dlg.exec_():
+            self.update_combobox()
+            # The +1 is needed because of the separator in the combobox
+            index = (names + custom_names).index(custom_name) + 1
+            self.schemes_combobox.setCurrentIndex(index)
+        else:
+            # Delete the config ....
+            custom_names.remove(custom_name)
+            self.set_option('custom_names', custom_names)
+            dlg.delete_color_scheme_stack(custom_name)
+
+    def edit_scheme(self):
+        """Edit current scheme."""
+        dlg = self.scheme_editor_dialog
+        dlg.set_scheme(self.current_scheme)
+
+        if dlg.exec_():
+            # Update temp scheme to reflect instant edits on the preview
+            temporal_color_scheme = dlg.get_edited_color_scheme()
+            for key in temporal_color_scheme:
+                option = "temp/{0}".format(key)
+                value = temporal_color_scheme[key]
+                self.set_option(option, value)
+            self.update_preview(scheme_name='temp')
+
+    def delete_scheme(self):
+        """Deletes the currently selected custom color scheme."""
+        scheme_name = self.current_scheme
+
+        # Put the combobox in Spyder by default, when deleting a scheme
+        names = self.get_option('names')
+        self.set_scheme('spyder')
+        self.schemes_combobox.setCurrentIndex(names.index('spyder'))
+        self.set_option('selected', 'spyder')
+
+        # Delete from custom_names
+        custom_names = self.get_option('custom_names', [])
+        if scheme_name in custom_names:
+            custom_names.remove(scheme_name)
+        self.set_option('custom_names', custom_names)
+
+        # Delete config options
+        for key in syntaxhighlighters.COLOR_SCHEME_KEYS:
+            option = "{0}/{1}".format(scheme_name, key)
+            CONF.remove_option(self.CONF_SECTION, option)
+        CONF.remove_option(self.CONF_SECTION, "{0}/name".format(scheme_name))
+
+        self.update_combobox()
+        self.update_preview()
+
+    def set_scheme(self, scheme_name):
+        """
+        Set the current stack in the dialog to the scheme with 'schema_name'.
+        """
+        dlg = self.scheme_editor_dialog
+        dlg.set_scheme(scheme_name)
+
+    @Slot(str)
+    def reset_to_default(self):
+        """Restore initial values for default color schemes."""
+        # Checks that this is indeed a default scheme
+        scheme = self.current_scheme
+        names = self.get_option('names')
+        if scheme in names:
+            for key in syntaxhighlighters.COLOR_SCHEME_KEYS:
+                option = "{0}/{1}".format(scheme, key)
+                value = CONF.get_default(self.CONF_SECTION, option)
+                self.set_option(option, value)
+
+            option = "{0}/name".format(scheme)
+            value = CONF.get_default(self.CONF_SECTION, option)
+            self.set_option(option, value)

```

Comment:
Is really necessary to restore the scheme name too? Since now users are not allowed to change names for default schemes, I'd say these three lines of code are not necessary :-)